blackwidowpluginbukkit/src/main/java/io/github/arcaneplugins/blackwidow/plugin/bukkit/logic/VersionInfo.java

```diff
@@ -0,0 +1,102 @@
+package io.github.arcaneplugins.blackwidow.plugin.bukkit.logic;
+
+import io.github.arcaneplugins.blackwidow.plugin.bukkit.util.ClassUtil;
+import io.github.arcaneplugins.blackwidow.plugin.bukkit.util.StringUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.InvalidObjectException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A custom implementation for comparing program versions
+ *
+ * @author stumper66
+ * @since 1.1.0
+ */
+public final class VersionInfo implements Comparable<VersionInfo> {
+
+    public VersionInfo(
+            @NotNull final String verStr
+    ) throws InvalidObjectException {
+        Objects.requireNonNull(verStr, "verStr");
+
+        this.verStr = verStr;
+
+        for (final String numTemp : verStr.split("\\.")) {
```

Comment:
I need to define a formal version specification for BlackWidow and make sure we can parse it with this update checker.